src/mui/list/DatagridBody.js

```diff
@@ -29,6 +31,9 @@ const DatagridBody = ({
                 selectable={false}
                 {...rowOptions}
             >
+                <TableRowColumn style={styles.cell['td:first-child']}>
+                    <Checkbox onCheck={(event, isInputChecked) => onSelectionChange(resource, id, isInputChecked)} />
```

Comment:
Extract the Checkbox into a new component. This component should extend `Component` and have a `handleCheck` method to avoid declaring a new function inside the jsx at each render